include/swift/AST/DiagnosticsCommon.def

```diff
@@ -71,6 +71,13 @@ ERROR(class_var_not_in_class,none,
       "class properties are only allowed within classes; "
       "use 'static' to declare a static property", ())
 
+ERROR(class_func_not_in_class_protocol,none,
+      "class methods are only allowed within classes; "
+      "use 'static' to declare a requirement fulfilled by either a class or static method", ())
```

Comment:
When diagnostics are related by only a snippet of text, as this one is to its neighbors to the north, you should parametrize the diagnostic.  [For example](https://github.com/apple/swift/blob/master/include/swift/AST/DiagnosticsSema.def#L305-L306)